plugins/destination/clickhouse/queries/table.go

```diff
@@ -56,3 +57,19 @@ func tableNamePart(table, cluster string) string {
 func GetPartitionKeyAndSortingKeyQuery(database, table string) string {
 	return fmt.Sprintf(`SELECT partition_key, sorting_key FROM system.tables WHERE database = '%s' AND name = '%s'`, database, table)
 }
+
+func GetTTLQuery(database, table string) string {
+	return fmt.Sprintf(`SHOW CREATE TABLE "%s"."%s"`, database, table)
+}
+
+func EqualTTLsQuery(table *schema.Table, ttl1, ttl2 string) string {
+	// ClickHouse allows different syntaxes for the same TTL expression,
+	// so we use a query to compare two given TTLs.
+	// However, to evaluate the expression, we replace referenced columns with a fixed date.
+	for i, col := range table.Columns {
+		// Replace column names with a fixed date to ensure the comparison works
+		ttl1 = strings.ReplaceAll(ttl1, col.Name, fmt.Sprintf("makeDate(1970, 1, 1) + INTERVAL %d SECOND", i))
```

Comment:
nit: Since it's just a string comparison we don't really need `makeDate`, `now64()` would work as well.

Reply:
Yeah, string comparison wasn't the right word. As long as it's a constant expression it should work, and `now64()` is stable across invocations in a single query. But yes, `makeDate` is more explicitly constant.